internal/controllers/clusterrequest_controller.go

```diff
@@ -236,25 +236,28 @@ func (t *clusterRequestReconcilerTask) run(ctx context.Context) (ctrl.Result, er
 		// after hw plugin is ready
 
 		// Handle the cluster install with ClusterInstance
-		err = t.handleClusterInstallation(ctx, renderedClusterInstance)
+		err := t.handleClusterInstallation(ctx, renderedClusterInstance)
 		if err != nil {
 			return requeueWithError(err)
 		}
 	}
 
 	// Handle policy configuration only after the cluster provisioning
-	// has started.
-	crProvisionedCond := meta.FindStatusCondition(
-		t.object.Status.Conditions,
-		string(utils.CRconditionTypes.ClusterProvisioned))
-	if crProvisionedCond != nil {
+	// has started, and not failed or timedout (completed, in-progress or unknown)
+	if utils.IsClusterProvisionStarted(t.object) &&
+		!utils.IsClusterProvisionFailed(t.object) &&
+		!utils.IsClusterProvisionTimedOut(t.object) {
```

Comment:
It might be better to have these checks in their own function, `isClusterReadyForConfig`, since they're used exactly the same below. We would avoid issues in case just one place is mistakenly updated.